src/Json.Schema.ToDotNet/ClassGenerator.cs

```diff
@@ -381,6 +402,95 @@ private ConstructorDeclarationSyntax GenerateDefaultConstructor()
                             SuffixedTypeName)));
         }
 
+        /// <summary>
+        /// Generates an initialization statement for each property for which the
+        /// schema specifies a default value.
+        /// </summary>
+        /// <remarks>
+        /// The resulting statements are inserted into the default constructor.
+        /// This ensures that the default values are set even if the object is
+        /// not the result of deserializing a JSON instance document.
+        /// <remarks>
+        /// <returns>
+        /// An array containing one initialization statement for each property
+        /// for which the schema specifies a default value.
+        /// </returns>
+        private ExpressionStatementSyntax[] GenerateDefaultInitializations()
+        {
+            var initializations = new List<ExpressionStatementSyntax>();
+
+            foreach (string propertyName in PropInfoDictionary.GetPropertyNames())
+            {
+                if (IncludeProperty(propertyName))
+                {
+                    PropertyInfo propInfo = PropInfoDictionary[propertyName];
+                    object defaultValue = propInfo.DefaultValue;
+
+                    ExpressionStatementSyntax initializationStatement = GenerateDefaultInitialization(propertyName, defaultValue);
+                    if (initializationStatement != null)
+                    {
+                        initializations.Add(GenerateDefaultInitialization(propertyName, defaultValue));
+                    }
+                }
+            }
+
+            return initializations.ToArray();
+        }
+
+        private ExpressionStatementSyntax GenerateDefaultInitialization(
+            string propertyName,
+            object defaultValue)
+        {
+            LiteralExpressionSyntax defaultValueExpression = GetLiteralExpressionForValue(defaultValue);
+            if (defaultValueExpression == null) { return null; }
+
+            return SyntaxFactory.ExpressionStatement(
+                SyntaxFactory.AssignmentExpression(
+                    SyntaxKind.SimpleAssignmentExpression,
+                    SyntaxFactory.IdentifierName(propertyName),
+                    defaultValueExpression));
+        }
+
+        private LiteralExpressionSyntax GetLiteralExpressionForValue(object value)
+        {
+            LiteralExpressionSyntax literalExpression = null;
+
+            if (value is bool)
+            {
+                SyntaxKind literalSyntaxKind = (bool)value == true
```

Comment:
>== true [](start = 59, length = 7)

you don't actually need the '== true; here, the bool is sufficient to drive the ternary conditional

Reply:
I know, I waffled on that, but I decided to write it this way because of the way it reads in English: "If the value is `true`, use a `TrueLiteralExpression`, otherwise use a `FalseLiteralExpression`. The "`== true`" reads as "is true" in my head.

---
In reply to: [244416578](https://github.com/Microsoft/jschema/pull/60#discussion_r244416578) [](ancestors = 244416578)

Reply:
i don't know what it is but I mostly find that direct comparisons to bool literals compromise rather than improve readability. this one's a corner case. 

---
In reply to: [244417679](https://github.com/Microsoft/jschema/pull/60#discussion_r244417679) [](ancestors = 244417679,244416578)